test/accounts.js

```diff
@@ -1,263 +1,195 @@
+    } else {
+      balanceInEther.toNumber();
+    }
 
-    return web3.eth
-      .sign("some data", secondAddress)
-      .then((result) => {
-        assert.fail("Expected an error while signing when not managing the private key");
-      })
-      .catch((err) => {
-        assert(err.message.toLowerCase().indexOf("cannot sign data; no private key") >= 0);
-      });
+    // Can't check the balance exactly. It cost some ether to send the transaction.
+    assert(balanceInEther > 4);
+    assert(balanceInEther < 5);
   });
 
-  it("should create a 2 accounts when passing an object to provider", function(done) {
-    var web3 = new Web3();
-    web3.setProvider(
-      Ganache.provider({
-        accounts: [{ balance: "0x12" }, { balance: "0x13" }]
-      })
-    );
-
-    web3.eth.getAccounts(function(err, result) {
-      if (err) {
-        return done(err);
-      }
-      assert(result.length, 2, "The number of accounts created should be 2");
-      done();
-    });
+  it("errors when we try to sign a transaction from an account we're impersonating", async() => {
+    const options = {
+      mnemonic,
+      secure: true,
+      unlocked_accounts: [0, badAddress]
+    };
+
+    const { web3 } = await initializeTestProvider(options);
+
+    try {
+      await web3.eth.sign("some data", badAddress);
+      assert.fail("Expected an error while signing when not managing the private key");
+    } catch (error) {
+      assert(error.message.toLowerCase().indexOf("cannot sign data; no private key") >= 0);
+    }
   });
 
-  it("should create a 7 accounts when ", function(done) {
-    var web3 = new Web3();
-    web3.setProvider(
-      Ganache.provider({
-        total_accounts: 7
-      })
-    );
-
-    web3.eth.getAccounts(function(err, result) {
-      if (err) {
-        return done(err);
-      }
-      assert(result.length, 7, "The number of accounts created should be 7");
-      done();
-    });
-  });
+  it("should create a 2 accounts when passing an object to provider", async() => {
+    const options = {
+      accounts: [{ balance: "0x12" }, { balance: "0x13" }]
+    };
 
-  it("should respect the default_balance_ether option", function(done) {
-    var web3 = new Web3();
-    web3.setProvider(
-      Ganache.provider({
-        default_balance_ether: 1.23456
-      })
-    );
+    const { accounts } = await initializeTestProvider(options);
 
-    web3.eth.getAccounts(function(err, accounts) {
-      if (err) {
-        return done(err);
-      }
-
-      function checkBalance(account) {
-        return new Promise(function(resolve, reject) {
-          web3.eth.getBalance(accounts[0], function(err, balance) {
-            if (err) {
-              return reject(err);
-            }
+    assert.strictEqual(accounts.length, 2, "The number of accounts created should be 2");
+  });
 
-            var balanceInEther = web3.utils.fromWei(balance, "Ether");
+  it("should create a 7 accounts when ", async() => {
```

Comment:
Can you update this so it says `"should create the correct number of accounts as specified by total_accounts"`  or something like it?